packages/fork-choice/src/forkChoice/forkChoice.ts

```diff
@@ -296,15 +296,14 @@ export class ForkChoice implements IForkChoice {
    *
    */
   predictProposerHead(headBlock: ProtoBlock, secFromSlot: number, currentSlot: Slot): ProtoBlock {
+    const {proposerBoost, proposerBoostReorg} = this.opts ?? {};
     // Skip re-org attempt if proposer boost (reorg) are disabled
-    if (!this.opts?.proposerBoost || !this.opts?.proposerBoostReorg) {
-      this.logger?.verbose("No proposer boot reorg prediction since the related flags are disabled");
-      return headBlock;
-    }
-
-    if (headBlock.slot === currentSlot) {
-      // If head block aka the head cache is current, that means `updateHead` is called during gossip handling,
-      // that can only happen if shouldOverrideForkChoiceUpdate = false so no reorg
```

Comment:
This check was added in https://github.com/ChainSafe/lodestar/pull/7810 but seems like an oversight as the comment is not correct

Reply:
yea this is not correct